### PR TITLE
Adding Copy and Move constructors for LSTM class

### PR DESCRIPTION
**Pull request by gauravghati**
This pull request is in reference to [issue#2625](https://github.com/mlpack/mlpack/issues/2625),
I'm trying to add constructors to LSTM layer

### CODE REVIEW

File: src/mlpack/methods/ann/layer/lstm_impl.hpp [hunk 1]
```diff
@@ -24,6 +24,90 @@ LSTM<InputDataType, OutputDataType>::LSTM()
...
+    bpttSteps(std::move(layer.bpttSteps))
+{
+  // Nothing to do here.
+}
+
+template <typename InputDataType, typename OutputDataType>
+LSTM<InputDataType, OutputDataType>& 
+LSTM<InputDataType, OutputDataType> :: operator=(const LSTM& layer)
+{
+    if(this != &layer)
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/lstm_impl.hpp:69]:
```suggestion
    if (this != &layer)
```

Minor style issue, missing space.

File: src/mlpack/methods/ann/layer/lstm_impl.hpp [hunk 2]
```diff
@@ -24,6 +24,90 @@ LSTM<InputDataType, OutputDataType>::LSTM()
...
+    {
+        inSize = layer.inSize;
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/lstm_impl.hpp:71]: The indentation is not quite right, should be one space, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#brace-placement for more details. That applies to each line.

File: src/mlpack/methods/ann/layer/lstm_impl.hpp [hunk 3]
```diff
@@ -24,6 +24,90 @@ LSTM<InputDataType, OutputDataType>::LSTM()
...
+        bpttSteps = layer.bpttSteps;
+    }
+    return *this; 
+}
+
+template <typename InputDataType, typename OutputDataType>
+LSTM<InputDataType, OutputDataType>& 
+LSTM<InputDataType, OutputDataType> :: operator=(LSTM&& layer)
+{
+    if(this != &layer)
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/lstm_impl.hpp:92]:
```suggestion
    if (this != &layer)
```

Minor style issue, missing space.

File: src/mlpack/tests/ann_layer_test.cpp [hunk 4]
```diff
@@ -1263,6 +1263,40 @@ TEST_CASE("CheckCopyMoveFastLSTMTest", "[ANNLayerTest]")
   CheckRNNMoveFunction<>(model2, input, target, 1);
 }
 
+/**
+ * Check whether copying and moving network with LSTM is working or not.
+ */
+TEST_CASE("CheckCopyMoveLSTMTest", "[ANNLayerTest]")
```
Comment (zoq) [inline src/mlpack/tests/ann_layer_test.cpp:1269]: Looks great to me.

File: src/mlpack/methods/ann/layer/lstm_impl.hpp (hunk 3 above)
Comment (gauravghati) [reply src/mlpack/methods/ann/layer/lstm_impl.hpp]: done changes!

File: src/mlpack/methods/ann/layer/lstm_impl.hpp (hunk 2 above)
Comment (gauravghati) [reply src/mlpack/methods/ann/layer/lstm_impl.hpp]: yeah, done.

**Review by zoq — APPROVED** [review]
Looks good to me, thanks for putting this together.

**Comment by gauravghati** [comment]
@zoq, Could you please merge this PR?

**Comment by zoq** [comment]
> @zoq, Could you please merge this PR?

Absolutely, thanks again for the contribution.

**Pull request merged**